api/api_sources/sources/libs/core-database/base.data.controller.ts

```diff
@@ -45,6 +45,7 @@ export interface BaseDataController {
     dependencies: any[];
     meta: ControllerMetaData;
     idKey: string;
+    repo: Repository<any>;
```

Comment:
This a functional interface for API, the repo property already available for actual BaseDataController class implementation.